src/registry/client.go

```diff
@@ -27,7 +27,26 @@ const (
 	VersionLabel = "com.dynatrace.build-version"
 )
 
-func (r *Client) GetImageVersion(ctx context.Context, ref name.Reference, keychain authn.Keychain, transport *http.Transport) (ImageVersion, error) {
+type Builder struct {
+	client Client
+}
+
+func NewClientBuilder() Builder {
+	return Builder{
+		client: Client{},
+	}
+}
+
+func (builder Builder) Build() Client {
+	return builder.client
+}
```

Comment:
just this is enough: (constructor != builder-pattern)
```
func NewClient() *Client {
	return &Client{}
}
```

Lets not over do it 😅

Reply:
True :+1: [Fixed](https://github.com/Dynatrace/dynatrace-operator/pull/1998/commits/d46e3d260e9902322dde894eb54583e4d6a318b7)